config/tgl-cavs.toml

```diff
@@ -0,0 +1,196 @@
+version = [2, 6]
+
+[adsp]
+name = "tgl"
+machine_id = 10
+image_size = "0x100000"
+
+[[adsp.mem_zone]]
+type = "ROM"
+base = "0x9F180000"
+size = "0x00002000"
+[[adsp.mem_zone]]
+type = "IMR"
+base = "0xB0000000"
+size = "0x1000000"
+[[adsp.mem_zone]]
+type = "HP-SRAM"
+base = "0xBE000000"
+size = "0x800000"
+[[adsp.mem_zone]]
+type = "LP-SRAM"
+base = "0xBE800000"
+size = "0x40"
+
+[cse]
+partition_name = "ADSP"
+[[cse.entry]]
+name = "ADSP.man"
+offset = "0x5c"
+length = "0x464"
+[[cse.entry]]
+name = "cavs0015.met"
+offset = "0x4c0"
+length = "0x70"
+[[cse.entry]]
+name = "cavs0015"
+offset = "0x540"
+length = "0x0"  # calculated by rimage
+
+[css]
+
+[signed_pkg]
+name = "ADSP"
+[[signed_pkg.module]]
+name = "cavs0015.met"
+
+[adsp_file]
+[[adsp_file.comp]]
+base_offset = "0x2000"
+
+[fw_desc.header]
+name = "ADSPFW"
+load_offset = "0x30000"
+
+[module]
```

Comment:
It would be a lot better if we could reuse the existing machine definitions and #include the extended manifest e.g.
```
[module]
file = "BRNGUP.toml"
file = "MIXIN.toml"
```

Reply:
let me check toml

Reply:
And we still need to reform existing machine definitions if toml is used since the definition format is not compatible with toml

Reply:
Another way without modifying existing machine definitions: use our own code to decode these config files. I first implemented some functions to achieve this, but at last I decided to use current toml tools.  @lgirdwood what's your idea ?